applications/FluidDynamicsApplication/python_scripts/fluid_rve_analysis.py

```diff
@@ -0,0 +1,203 @@
+        self.boundary_mp_name = project_parameters["rve_settings"]["boundary_mp_name"].GetString()
+        self.averaging_mp_name = project_parameters["rve_settings"]["averaging_mp_name"].GetString()
+        self.print_rve_post = project_parameters["rve_settings"]["print_rve_post"].GetBool()   
+
+        self.averaging_volume = -1.0  # it will be computed in initialize
+        self.domain_size = project_parameters["solver_settings"]["domain_size"].GetInt()
+        if self.domain_size != 2 and self.domain_size != 3:
+            err_msg = "Wrong 'domain_size' value {}. Expected 2 or 3.".format(self.domain_size)
+            raise ValueError(err_msg)
+
+        # Dictionary containin the jump data for periodicity
+        self.strains = {}
+        
+        self.strains["00"] = project_parameters["rve_settings"]["jump_XX"].GetDouble()
+        self.strains["11"] = project_parameters["rve_settings"]["jump_YY"].GetDouble()
+        self.strains["01"] = project_parameters["rve_settings"]["jump_XY"].GetDouble()
+        if(self.domain_size == 3):
+            self.strains["22"] = project_parameters["rve_settings"]["jump_ZZ"].GetDouble()
+            self.strains["02"] = project_parameters["rve_settings"]["jump_XZ"].GetDouble()
+            self.strains["12"] = project_parameters["rve_settings"]["jump_YZ"].GetDouble()
+            
+        self.strain = KratosMultiphysics.Matrix(self.domain_size, self.domain_size)
+        self.strain.fill(0.0)
+        
+        # Create and fill "strain" matrix
+        for i in range(self.domain_size):
+            for j in range(i ,self.domain_size):
+                strainIndex = str(i) + str(j)
+                
+                self.strain[i, j] = self.strains[strainIndex]
+                self.strain[j, i] = self.strains[strainIndex]
+
+        self.populate_search_eps = 1e-4 ##tolerance in finding which conditions belong to the surface (will be multiplied by the lenght of the diagonal)
+        self.geometrical_search_tolerance = 1e-4 #tolerance to be used in the search of the condition it falls into
+
+        super().__init__(model, project_parameters)
+
+    # Here populate the submodelparts to be used for periodicity
+    def ModifyInitialGeometry(self):
+        super().ModifyInitialGeometry()
+        boundary_mp = self.model[self.boundary_mp_name]
+        self.averaging_mp = self.model[self.averaging_mp_name]
+
+        # Construct auxiliary modelparts
+        self.min_corner, self.max_corner = self._DetectBoundingBox(self.averaging_mp)
+        self._ConstructFaceModelParts(self.min_corner, self.max_corner, boundary_mp)
+
+        self.averaging_volume = 1.0
+        for i in range(self.domain_size):
+            self.averaging_volume *= self.max_corner[i]-self.min_corner[i]
+        KratosMultiphysics.Logger.PrintInfo(self._GetSimulationName(), "RVE undeformed averaging volume = ", self.averaging_volume)
+
+    def ModifyAfterSolverInitialize(self):                
```

Comment:
```suggestion
    def ModifyAfterSolverInitialize(self):
        super().ModifyAfterSolverInitialize()
                        
```
Although it will always be potentially empty, I'd call the base class  `ModifyAfterSolverInitialize` (just in case we change the base class in the future).

Reply:
Ok, added.